### PR TITLE
Bump to Controller runtime 0.6.2

### PR DESCRIPTION
**Pull request by ChunyiLyu**
**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes

- bump to controller runtime `0.6.2`
- k8s.io api, apimachinery, and client-go bumps to `0.18.6
- some changes in the CRD spec since k8s api is updated
- fix a small logging msg bug as discussed in [here](https://vmware.slack.com/archives/CJWJ6U9GR/p1598536459182300?thread_ts=1598534350.177600&cid=CJWJ6U9GR)
- shortened couple methods names in system tests. names were too verbose

There are numbers of breaking changes coming in as this bump:

- new version of the envtest from controller-runtime expects namespaces were created before creating any resources in them. I refactored the integration tests that all tests are creating rmq instances in the "default" namespace, which always exists and we don't need to bother to create it.
- clientSet calls (Get, Create, Delete) need to take in `context.Context`

## Local Testing

Have run unit, integration, and system tests again `dev-bunny`

### CODE REVIEW

**Review by ansd — APPROVED** [review]

File: system_tests/system_tests.go [hunk 1]
```diff
@@ -255,22 +256,22 @@ CONSOLE_LOG=new`
 
 		It("persists messages after pod deletion", func() {
 			By("publishing a message", func() {
-				err := rabbitmqPublishToNewQueue(hostname, port, username, password)
+				err := publishToNewQueue(hostname, port, username, password)
```
Comment (harshac) [inline system_tests/system_tests.go:259]: Small: Rename to `publishToQueue`. I am assuming we are consuming from the same queue in the next `By` block.

File: system_tests/system_tests.go [hunk 2]
```diff
@@ -71,7 +72,7 @@ var _ = Describe("Operator", func() {
 
 		It("works", func() {
 			By("being able to create a test queue and publish a message", func() {
-				response, err := rabbitmqAlivenessTest(hostname, port, username, password)
+				response, err := alivenessTest(hostname, port, username, password)
```
Comment (harshac) [inline system_tests/system_tests.go:75]: Should we rename this to livenessTest to be consistent with the k8s terminology?

File: system_tests/system_tests.go [hunk 3]
```diff
@@ -355,14 +356,14 @@ CONSOLE_LOG=new`
...
 			It("talks amqps with RabbitMQ", func() {
 				var err error
-				username, password, err = getRabbitmqUsernameAndPassword(clientSet, "rabbitmq-system", "tls-test-rabbit")
+				username, password, err = getUsernameAndPassword(ctx, clientSet, "rabbitmq-system", "tls-test-rabbit")
 				Expect(err).NotTo(HaveOccurred())
 
 				// try to publish and consume a message on a amqps url
 				sentMessage := "Hello Rabbitmq!"
-				Expect(rabbitmqAMQPSPublishToNewQueue(sentMessage, username, password, hostname, amqpsNodePort, caFilePath)).To(Succeed())
+				Expect(publishToNewQueueAMQPS(sentMessage, username, password, hostname, amqpsNodePort, caFilePath)).To(Succeed())
```
Comment (harshac) [inline system_tests/system_tests.go:364]: Same as above `publishToQueueViaAMQPS`?

Comment (harshac) [inline system_tests/system_tests.go:364]: Same as above `publishToQueueViaAMQPS`?

File: system_tests/utils.go [hunk 4]
```diff
@@ -162,7 +162,7 @@ type Message struct {
 	MessageCount int    `json:"message_count"`
 }
 
-func rabbitmqPublishToNewQueue(rabbitmqHostName, rabbitmqPort, rabbitmqUsername, rabbitmqPassword string) error {
+func publishToNewQueue(rabbitmqHostName, rabbitmqPort, rabbitmqUsername, rabbitmqPassword string) error {
 	url := fmt.Sprintf("http://%s:%s/api/queues/%%2F/test-queue", rabbitmqHostName, rabbitmqPort)
```
Comment (harshac) [inline system_tests/utils.go:166]: Should we extract out test-queue as it's being used at several places? Upto you as it's only in the helpers.

File: system_tests/utils.go [hunk 5]
```diff
@@ -262,7 +262,7 @@ func rabbitmqAMQPSPublishToNewQueue(message, username, password, hostname, amqps
 	return nil
 }
 
-func rabbitmqAMQPSGetMessageFromQueue(username, password, hostname, amqpsPort, caFilePath string) (string, error) {
+func getMessageFromQueueAMQPS(username, password, hostname, amqpsPort, caFilePath string) (string, error) {
```
Comment (harshac) [inline system_tests/utils.go:265]: If we change the publish function to as suggested, change this to be consistent `getMessageFromQueueViaAMQPS`

**Review by harshac — CHANGES_REQUESTED** [review]

File: system_tests/system_tests.go (hunk 2 above)
Comment (ChunyiLyu) [reply system_tests/system_tests.go:75]: aliveness test refers to rabbitmq aliveness test via management api `/api/aliveness-test/`. I think the name is fine here.

File: system_tests/system_tests.go (hunk 3 above)
Comment (ChunyiLyu) [reply system_tests/system_tests.go:364]: Updated to `publishToQueueAMQPS`. Thanks!

File: system_tests/utils.go (hunk 5 above)
Comment (ChunyiLyu) [reply system_tests/utils.go:265]: I like `getMessageFromQueueAMQPS ` since it's shorter 😆

File: system_tests/utils.go (hunk 4 above)
Comment (ChunyiLyu) [reply system_tests/utils.go:166]: I will leave it like that for now. Will address it in a different PR, I have a list of small fixes anyways. Good point!

File: system_tests/system_tests.go (hunk 3 above)
Comment (ChunyiLyu) [reply system_tests/system_tests.go]: changed to `publishToQueueAMQPS` since it's shorter.

**Review by harshac — APPROVED** [review]

**Pull request merged**